ui/components/src/model/legend.ts

```diff
@@ -24,15 +24,23 @@ export interface LegendOptions {
 export interface LegendItem {
   id: string;
   label: string;
-  isSelected: boolean;
   color: string;
-  onClick: MouseEventHandler<HTMLElement>;
+  onClick?: MouseEventHandler<HTMLElement>;
```

Comment:
Making this optional because it is no longer required to manage the legend selection. Now it's an optional handler in case you want to do something extra when a legend item is clicked.

Reply:
Nice, in the future I could definitely see an onHover being useful here too for dispatching that chart highlight action:
```typescript
chart.dispatchAction({
  type: 'highlight',
  seriesIndex: seriesIndex, // legend item being hovered over
});

Reply:
💯 Implementing that is on my list of things to do for the table legend, so I'll likely add an `onHover` or something like that in the relatively near future.